### ISSUE TITLE
Make Java projects target Java 11

### ISSUE THREAD

**Issue by jabrah** [eclipse-pass/main#278]
Follow-on to #241, which laid the groundwork by ensuring that all relevant projects were compatible with JDK11 -- that the projects could build and run with JDK11, but still targeting Java 8. A couple of key issues were identified that prevented the updates of targeting Java 11, namely dependencies within Docker images and inter-project Docker dependencies.

Some projects built Docker images that were based on `openjdk:8x`. When feeding code compiled to Java 11, these images would immediately fail because the runtime environment was unable to read the bytecode.

Docker images that likely need updating (https://github.com/eclipse-pass/main/issues/270):

* `pass-indexer`
* `pass-notification-services`
* `pass-doi-service`
* `pass-deposit-services`
* `pass-authz/pass-authz-listener`: built from `pass-docker/authz/Dockerfile`

When ready, we can revive and merge these PRs to start targeting Java 11:

* https://github.com/eclipse-pass/pass-java-client/pull/181
* https://github.com/eclipse-pass/pass-deposit-services/pull/279
* https://github.com/eclipse-pass/pass-indexer/pull/34

In order to target Java 11, we should specify the release in the maven-compiler-plugin's configuration. This would ideally be done in the project parent POM, allowing projects that need to override the setting.

``` xml
from:
<configuration>
  <source>1.8</source>
  <target>1.8</target>
</configuration>

to:
<configuration>
  <release>11</release>
</configuration>
```

I personally think it would be a good idea to update Docker images to use JDK11 _first_, then update the projects to target Java 11. This should simplify the update process, especially around the use of Docker images in ITs.

****************************

### Incompatibilities

Because of the use of Fedora 4.7.5 as the basis for our `fcrepo` Docker image and the fact that we have no current plans to update this, any artifacts that are bundled in the `fcrepo` Docker image must target Java 8. 

* `pass-fcrepo-jms`
* `pass-fcrepo-jsonld`
* `pass-fcrepo-module-auth-rbacl`
* `pass-authz`
  * Except `pass-authz/pass-authz-listener`, which is deployed in a stand-alone image

**Comment by jabrah** [comment]
Bumping `maven-compiler-plugin` to compile to Java 11 included in POM cleanup #439

**Comment by jabrah** [comment]
`main` POM now configured to Java 11 (https://github.com/eclipse-pass/main/blob/main/pom.xml#L128-L129) with SNAPSHOT published to Sonatype to be used by child projects
